### PR TITLE
CI/tests: cache reference metrics | PoC w/ `text`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

it is a subset of #2335 with setting the CI and validation on Text only
with local runs, it seems to cut 40% times so for example BLUE score went from 2m32 to 1m24

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2382.org.readthedocs.build/en/2382/

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
@stancld is it possible that some metrics/references are not deterministic?

**Comment by Borda** [comment]
seems to be cache issue: https://github.com/python-cachier/cachier/pull/196

**Comment by Borda** [comment]
seems like unrelated GPU falling on GPU for timeout
```
FAILED unittests/image/test_fid.py::test_fid_same_input[768] - Failed: Timeout >240.0s
FAILED unittests/image/test_fid.py::test_fid_same_input[2048] - Failed: Timeout >240.0s
FAILED unittests/image/test_mifid.py::test_fid_same_input[2048] - Failed: Timeout >240.0s
```

File: .azure/gpu-unittests.yml [hunk 1]
```diff
@@ -148,8 +150,7 @@ jobs:
       - bash: |
           python -m pytest $(TEST_DIRS) -v \
             -m "not DDP" --numprocesses=5 --dist=loadfile \
-            --cov=torchmetrics --timeout=240 --durations=100 \
-            --reruns 3 --reruns-delay 1
+            --cov=torchmetrics --timeout=240 --durations=500
```
Comment (Borda) [inline .azure/gpu-unittests.yml:153]:
```suggestion
            --cov=torchmetrics --timeout=240 --durations=100 \
            --reruns 3 --reruns-delay 1
```

**Comment by Borda** [comment]
Observed speed-up for standard unittests on GPU is 34min -> 18min

**Pull request merged**